File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(a, b) {
    const map = new Map();
    for (let i = 0; i < a.length; i++) {
        let diff = b - a[i];
        if (map.has(diff)) {
            return [map.get(diff), i];
        }
        map.set(a[i], i);
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)

    for i, num :=  range nums{
        c := target - num

        if idx, ok := m[c]; ok{
            return []int{idx, i}
        }
        m[num] = i
    }
    return []int{}
}